dbflow/src/main/java/com/raizlabs/android/dbflow/sql/saveable/ModelSaver.java

```diff
@@ -113,4 +119,13 @@ boolean delete(ModelAdapter<TModel> modelAdapter, TAdapter adapter, TTable model
         adapter.updateAutoIncrement(model, 0);
         return successful;
     }
+
+    private void closeInsertStatementAsync(final DatabaseStatement insertStatement) {
+        CLEAN_UP_HANDLER.post(new Runnable() {
+            @Override
+            public void run() {
+                insertStatement.close();
```

Comment:
we should also reset the insertStatement in the `ModelAdapter` associated in this class. something like `clearInsertStatement()` should do the trick. We don't want to expose a setter which could lead to improper replacing :)